### ISSUE TITLE
No plot written in nfo file

### ISSUE THREAD

**Issue by mapleleafs89** [tinyMediaManager/tinyMediaManager#208]
Version: 2.9.4
Build: 2017-08-25 08:15
OS: Linux 3.10.0-514.16.1.el7.x86_64
JDK: 1.8.0_131 amd64 Oracle Corporation

__What is the actual behaviour?__
Plot stopped appearing in nfo file scraped

__What is the expected behaviour?__
Plot information exist

__Steps to reproduce:__
Scrape a movie (I tried several)
[tmm_logs.zip](https://github.com/tinyMediaManager/tinyMediaManager/files/1289263/tmm_logs.zip)

__Additional__
Have you attached the logfile from the day it happened?
Yes

**Comment by ProIcons** [comment]
Hello i would like to verify some more information before i proceed resolving your issue. Is this occurred in movie with title "It comes at Night" ? Or it occurred on other movies too?

**Comment by mrdwab** [comment]
I've been having the same problem the last day or two specifically with scraping movies from IMDB. Other scrapers are picking up the plot.

**Comment by ProIcons** [comment]
Ok. The way tmm works is as follows. TMM contacts each scraper to fetch a movie from the web. When the movie is fetched data returns back to tmm. And from tmm data goes into the nfo file. Since the plot is present on the GUI but not in the nfo indicating issue with i/o operations. And that's confirmed by the log file. TMM was unable to wrote nfo and got an Access Denied errror. Now there are several reasons that this issue may occur. First of all file was locked by another application, you don't have permissions to modify this file, you don't have permission to modify any file on this folder.

**Comment by mapleleafs89** [comment]
Hi Prolcons, thanks for your reply (and mrdwab too). I believe he is right and it is imdb where the issue occurs, I tested scraping manually using moviedb on "It comes at night" and the plot showed in the GUI, but as you have gathered from the logs I suppose there was a permissions issue writing over the current nfo (but I suppose it would have  worked otherwise). All other movies plot scraping are not working for past few days. Thanks for verifying there will be an upcoming fix for this issue, appreciate it!!

**Comment by ProIcons** [comment]
Github changed it's structure. So out scraper broke. There will be a new hotfix asap.

**Comment by R1CH-T** [comment]
I just downloaded the latest nightly build for Linux ([tmm_2.9.5-SNAPSHOT_5f779ae_linux.tar.gz](http://nightly.tinymediamanager.org/dist/tmm_2.9.5-SNAPSHOT_5f779ae_linux.tar.gz)), backed up the old scraper plug-in and then extracted `scraper-imdb.jar` to the `plugins` directory of my TMM (v. 2.9.4).

It works flawlessly.

This needs to be pushed out ASAP.

IMDB may be technically inferior to TMDB, but the ratings are a very important factor for viewing choice and IMDB has the sheer weight of numbers in it's favour.

Perhaps IMDB ratings could be selected for in the UI?

Thanks for everything.
Rich. :)

**Comment by R1CH-T** [comment]
By the way: Before trying this out, I switched to TMDB and it scraped fine, but the ratings were all showing "0".

The TMDB plugin might also have a problem.

I would recommend further testing.

Thanks,

Rich.

**Comment by myron0815** [comment]
TMDB ratings work for me - which movie did you try?

**Comment by ProIcons** [comment]
TMDb Service hasn't changed it's structure, and therefore ratings haven't been affected since the TMM's version 2.9.3. Having said that, i would really like to examine your log file to determine if there's an issue after all on TMDb's scraper. Note that Kodi's TMDb scraper and TMM's TMDb scraper differs. There are 2 completely different scrapers, and they don't work the same way. 
Is known that Kodi's scrapers on TMM are having some small issues, depending on your Kodi version.

**Comment by mapleleafs89** [comment]
Yes agreed with R1CH-T, ratings are better sourced from imdb which is why I use it. Any idea when the hotfix for IMDb will make it to public build?

**Comment by ProIcons** [comment]
In a week or so

**Comment by R1CH-T** [comment]
Hey guys,

Sorry for the delay in replying.

I just tested the TMDB scraper and it worked just fine, scraping all data including the ratings. :)

I was able to replicate the issue by selecting "Kodi: The Movie Database - 3.9.5" from the movie scraper settings. This must have been the scraper i used before. The internal scraper "themoviedb.org" worked perfectly.

Where I said above:

> Perhaps IMDB ratings could be selected for in the UI?

I had forgotten this option already exists, so sorry about that.

Interestingly, when i choose either IMDB or TMDB from this "Get Rating from" scraper option using the Kodi version of the TMDB scraper, I get a zero rating from both.

Here is a log file for this situation:
[tmm.log](https://github.com/tinyMediaManager/tinyMediaManager/files/1349069/tmm.log)

Thanks,
Rich.
